_observing-your-data/alerting/triggers.md

```diff
@@ -125,15 +125,76 @@ Variable | Data type | Description
+`bucket_keys` | String | Comma-separated list of the monitor's bucket key values. Available only for `ctx.dedupedAlerts`, `ctx.newAlerts`, and `ctx.completedAlerts`. Accessed through `ctx.dedupedAlerts.0.bucket_keys`.
+`parent_bucket_path` | String | The parent bucket path of the bucket that triggered the alert. Accessed through `ctx.dedupedAlerts.0.parent_bucket_path`.
+`associated_queries` | Array | An array of document-level monitor queries that triggered the creation of the finding associated with the alert. Only available with document-level monitors. Accessed through `ctx.alerts.0.associated_queries`.
+`sample_documents` | Array | An array of sample documents that matched the monitor query. Only available with bucket-level, and document-level monitors. Accessed through `ctx.newAlerts.0.sample_documents`, and `ctx.alerts.0.sample_documents`, respectively.
+
+#### `associated_queries`, and `sample_documents` variables
+
+Per bucket, and per document monitors support printing sample documents in notification messages; and per document monitors support printing the list of queries that triggered the creation of the finding associated with the alert. Each new alert that's generated during a monitor execution will be added to a list (i.e., `newAlerts` for per bucket monitors, and `alerts` for per document monitors) within the `ctx` variable. Each alert has its own list of `sample_documents`, and each per document monitor alert has its own list of `associated_queries`. The message template can be formatted to iterate through the list of alerts, and the list of `associated_queries`, and `sample_documents` for each alert.
+
+To learn more about security in the Alerting plugin, see [Alerting security](https://opensearch.org/docs/latest/observing-your-data/alerting/security/).
+
+#### Sample documents variables
+
+Variable | Data type | Description
+:--- | :--- | : ---
+`_index` | String | The index containing the 
+`_id` | String | The document ID for the document in its index.
+`_score` | Float | A positive 32-bit floating point number illustrating the relevance of the returned document.
+`_source` | Object | The JSON payload of the sample document.
+
+##### Mustache template example
+
+```groovy
+Alerts:
+{{#ctx.alerts}}
+    RULES
+    {{#associated_queries}}
+        Name: {{name}}
+        Id: {{id}}
+        Tags: {{tags}}
+    ------------------------
+    {{/associated_queries}}
+{{/ctx.alerts}}
+```
+
+#### Associated queries variables
+
+Variable | Data type | Description
+:--- | :--- | : ---
+`id` | String | The ID for the document-level query.
+`name` | String | The name configured for the document-level query.
+`tags` | Array | An array of tags (each of type String) configured for the document-level query.
+
+##### Mustache template example
+
+**Note**: The `_source` object in this example is based on the `opensearch_dashboards_sample_data_ecommerce` index that's available on OpenSearch Dashboards.
+
+```groovy
+Alerts
+{{#ctx.alerts}}
```

Comment:
**Question for documentation reviewers:**
The following on line 146, calls out that the variable names for the "alerts" list is `newAlerts` for per bucket monitors, and `alerts` for per document monitors.
> Each new alert that's generated during a monitor execution will be added to a list (i.e., `newAlerts` for per bucket monitors, and `alerts` for per document monitors) within the `ctx` variable.

This example is for a per document monitor as the "alerts" list is access via `{{#ctx.alerts}}` instead of `{{#ctx.newAlerts}}`. Does that seem clear enough?

Reply:
Lets update line 184 and call it out.

Reply:
Revised.